docs/development/new-cloud-provider-api-extension.md

```diff
@@ -0,0 +1,79 @@
+    * Select the cloud provider in `DetermineCloudProviderInShoot()`
+    * Return the cloud provider in the error at the end of `DetermineCloudProviderInShoot()` if the number of clouds is not 1
+    * Return the appropriate image name for the cloud provider in `DetermineMachineImage()`
+    * Return the latest available kubernetes version in `DetermineLatestKubernetesVersion()` 
+8. Update [pkg/apis/garden/v1beta1/types.go](../../pkg/apis/garden/v1beta1/types.go) to include the cloud provider at the following key points:
+    * Define the name of the cloud provider as a key of `CloudProfileSpec struct`, where the value is a pointer to `<provider>Profile`. Be sure to include json tags on the struct definition.
+    * Define the struct for `<provider>Profile`. See the existing examples. The `Profile` contains all of the properties necessary to call the cloud provider, including constraints.
+9. Update [pkg/apis/garden/validation/validation.go](../../pkg/apis/garden/validation/validation.go) to include the cloud provider at the following key points:
+    * Include the optional cloud provider DNS, if defined, in `availableDNS` in `func init()`
+    * Include the cloud provider in the error message in `ValidateCloudProfileSpec()` if there was an error returned from `helper.DetermineCloudProviderInProfile`.
+    * Validate all of the necessary elements for the cloud provider in `ValidateCloudProfileSpec()`, and add all the errors to `allErrs`. See the existing examples.
+    * Write validation functions for the cloud provider for each of the elements validates in the previous step. See the existing examples.
+    * In `validateCloud()`, create validation logic for the cloud provider. See existing examples. The net result should be any errors add to `allErrs`.
+    * In `ValidateShootSpecUpdate()`, validate immutable fields are not changed. See existing examples.
+10. Create validation tests in [pkg/apis/garden/validation/validation_test.go](../../pkg/apis/garden/validation/validation_test.go) . See existing examples.
+11. Run [hack/generate-code](../../hack/generate-code) to generate the necessary code.
+
+## Add Validations
+
+Validations check that parameters submitted to Gardener for the given cloud profile are valid. It has two major steps:
+
+1. Write validators
+2. Write tests for the validators
+
+The details are as follows:
+
+1. Update [plugin/pkg/shoot/quotavalidator/admission.go](../../plugin/pkg/shoot/quotavalidator/admission.go) at the following key points:
+    * Add logic in `getShootWorkerResources()` to populate the return `[]quoteworker` slice with the workers for the cloud provider.
+    * Add logic in `getMachineTypes()` to return the list of machine types available for the cloud provider.
+    * Add logic in `getVolumeTypes()` to return the list of volume types available for the cloud provider.
+    * Add logic in `quotaVerificationNeeded()` to return if quote verification is required. Note: What is "quote verification"?
```

Comment:
s/quote/quota

It is possible to define Quotas for Shoot clusters, e.g., project "A" may use shoots with max. 8 CPUs.
The "quota verification" function checks whether a shoot that was put under quota has changes in relevant sections of the shoot specification that might be related to CPU (in this example), e.g. the worker pools. The admission controller checks whether this change is allowed or not (constrained by the defined Quota).